### PR TITLE
[guiinfo] New gui infolabel bool System.HasDisplayHdr and System.HdrIsEnabled

### PR DESCRIPTION
**Pull request by CastagnaIT**
## Description

Give to the add-ons a chance to know if the display have HDR capability and if is enabled

## Motivation and Context

in the addon where i'm working on i need to know if the display have HDR capablility
in order to automatically enable this feature on the service where i perform the requests

I was thinking of implementing it on the python object "xbmcgui",
then i thought that as an infolabel it might also be useful for other purposes

## How Has This Been Tested?

builded Kodi 19 in my machine and i saw the output of
xbmc.getCondVisibility('System.HasDisplayHdr')
xbmc.getCondVisibility('System.HdrIsEnabled')

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [x] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [x] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by lrusak** [comment]
What is the underlying reason for this change? Why would you need to know this information?

In my mind Kodi shouldn't care if the media is HDR or not and should just cope with the information it is given.

**Comment by da-anda** [comment]
not sure if infolabels are the proper place to expose this information. Can't you query the settings from an add-on to check if the HDR setting is enabled?

**Comment by CastagnaIT** [comment]
> Perché avete bisogno di conoscere queste informazioni?

In the case of Netflix, in order to get HDR video streams i have to make a precise request request to the netflix service.
But i have no way to know if the device actually supports HDR and if it is enabled in Kodi's options (also get settings by rpc not work, but prefer a more direct way instead use jsonrpc).

Until now, in order to circumvent this lack, it was necessary to add an appropriate setting in Netflix addon where an user can manually enable/disable hdr. But this it is not ideal because it can also be enabled on non HDR systems, or also activate it while actually in Kodi the HDR is not enabled.

Kodi already has these two data available, so having access, I can solve this problem of double settings, which actually can be used in inappropriate ways.

if using infolabels is not suitable, no problem i just need to know where to implement it, unless it gets too complex

**Comment by lrusak** [comment]
I don't think the design of the current Kodi internals are that great for HDR at this time.

I believe that Kodi should simply advertise if a windowing system supports HDR and that's it. That way any add-ons can enable/disable HDR media grabbing if desired. If this PR is the correct way to advertise that I can't really comment on.

**Comment by CastagnaIT** [comment]
I know that right now is not the best time,
then add only an option (instead of two), like this:
<pre>
case SYSTEM_HDR_IS_ENABLED:
    {
      value =  CServiceBroker::GetWinSystem()->IsHDRDisplay() && 
	           CServiceBroker::GetSettingsComponent()->GetSettings()->GetBool(CServiceBroker::GetWinSystem()->SETTING_WINSYSTEM_IS_HDR_DISPLAY)
      return true;
    }
</pre>

let me know news

**Comment by ksooo** [comment]
In my opinion (but maybe I'm wrong) using GUI info labels/bools/ints to control non-GUI addon functionality like assembling the "right" server requests is conceptually wrong. Although it might work, this is not what GUI labels/bools/ints are designed for.

IMO, what you are actually looking for is a new/extended json-rpc / python API extension.

**Comment by CastagnaIT** [comment]
What do you think if I add it inside the xbmcgui python object?  It could go well?

**Comment by ksooo** [comment]
> What do you think if I add it inside the xbmcgui python object?

 Sorry, I don't know anything about "the xbmcgui python object", but "gui" does not sound right...

**Comment by CastagnaIT** [comment]
now that I've read the description I guess it's not the exact place, it's probably better in the "xbmc" py object, there's a little bit of everything here:
https://romanvm.github.io/Kodistubs/_autosummary/xbmc.html

if possible, someone who eats python, let me know what the best way

i would like to avoid JSONRPC because it is only a true/false value, in my opinion i think there is only unnecessary overload to get only this simply value

File: xbmc/guilib/guiinfo/SystemGUIInfo.cpp [hunk 1]
```diff
@@ -651,6 +651,16 @@ bool CSystemGUIInfo::GetBool(bool& value, const CGUIListItem *gitem, int context
...
       break;
     }
+    case SYSTEM_HAS_DISPLAY_HDR:
+    {
+      value = CServiceBroker::GetWinSystem()->IsHDRDisplay();
+      return true;
+    }
+    case SYSTEM_HDR_IS_ENABLED:
+    {
+      value = CServiceBroker::GetSettingsComponent()->GetSettings()->GetBool(CServiceBroker::GetWinSystem()->SETTING_WINSYSTEM_IS_HDR_DISPLAY);
```
Comment (Rechi) [inline xbmc/guilib/guiinfo/SystemGUIInfo.cpp]: > The `ColumnLimit` in `.clang-format` is set to `100` which defines line length [...]

https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#3-formatting

**Review by Rechi — COMMENTED** [review]
Please follow the [current code guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md).

**Comment by CastagnaIT** [comment]
@Rechi this is not the definitive solution,
I'm waiting for someone to tell me how to proceed

**Comment by da-anda** [comment]
just a headsup: since peak3d also needs to expose some system info to python, we will first get his PR in shape so that everybody is happy and then you can add your info accordingly
see https://github.com/xbmc/xbmc/pull/17265

But in addition to the python API to get this information, IMO an infolabel/boolean that indicates that current output is HDR or that the system is HDR capable might probably not be a bad idea for skins to play with in conditions or maybe even to change the color of certain elements (dim down bright components etc). But this is just my opinion.

**Comment by CastagnaIT** [comment]
great i take your suggest, then summarizing:
I create only one new option instead two,
and i move the code under xbmc  `xbmc.isHdrEnabled`, like PR mentionated

last thing, you want to keep also the infolabel or not?
because i see conflicting opinions

**Comment by da-anda** [comment]
please wait with `xbmc.isHdrEnabled` until the other PR is merged, since we are currently discussing if we group system related information in `xbmc.system` which would be similar to JSON-RPC or handle it in some other way.
And the infolabel/bool is still in discussion. Opinions by skinners welcome though.

**Comment by RayDawGGG** [comment]
I apologize if this the incorrect place to discuss this. My inquiry is about the creation of Infolabels. HDR is a format and not a video codec or a video type. In order to use a Infolabel for HDR flags in skinning a new Infolabel would have to be created.As of now ListItem.FileName is used to falsely relay HDR format info in a skin. Is it possible to create a new Infolabel? Or are there already other ways to obtain HDR format info?

**Comment by CastagnaIT** [comment]
there are currently no other ways to obtain info
but this PR isn't exactly about what you want to do

this infolabel would only be used to know when HDR is enabled or not

But currently the HDR functionality on Kodi 19 needs to be rebuilt and revised, which seems to have been postponed, only when will be completed you will be able to add any type of these implementations

**Comment by matthuisman** [comment]
@CastagnaIT i have a few addons that would like to know when the screen is HDR / Dolby Vision compatible.
Any thoughts on re-doing this PR ?

**Comment by CastagnaIT** [comment]
for hdr yes, but i dont know if there is info about dolby vision in kodi core
yeah know this info would be very useful to automatize addons to query hdr/dv streams

we should add a new python module `xbmcsystem` to add system related things
i have not yet had time to study how to add the new module yet

**Pull request closed without merging**